parser/src/lib.rs

```
@@ -80,16 +80,48 @@ impl<'s> Parser<'s> {
     pub fn arg(&mut self) -> OptExpr {
         self.ident()
     }
+    pub fn expr(&mut self) -> ResultExpr {
+        match self.inner_assign() {
+            Ok(None) => self.ret(),
+            Ok(Some(x)) => Ok(x),
+            Err(x) => Err(x),
+        }
+    }
+    pub fn inner_assign(&mut self) -> ResultOptExpr {
+        let mutability = self.lexer.collect_of_if(&[Token::Let, Token::Const]);
+        if let Some(muta) = mutability {
+            let identifier = self
+                .ident()
+                .expect_expr("expected identifier".to_string())?;
+            // TODO:: Add various assignments
+            let _ = self
+                .lexer
+                .collect_of_if(&[Token::As])
+                .expect_token("expected =".to_string())?;
+            return self.or_cmp().convert_to_result_opt().result_opt_or(|asgn| {
+                self.lexer
+                    .collect_if(Token::SColon)
+                    .expect_token("expected ';'".to_string())?;
+                bubble_expr!(AsDef, muta, identifier, asgn)
+            });
+        }
+        Ok(None)
+    }
     pub fn block(&mut self) -> ResultExpr {
         self.lexer
             .collect_if(Token::OBrace)
             .expect_token("expected '{'".to_string())?;
-        self.ret().result_or(|expr| {
-            self.lexer
-                .collect_if(Token::CBrace)
-                .expect_token("expected '}'".to_string())?;
-            result_expr!(Block, vec![expr])
-        })
+        let mut exprs: Vec<Box<Expr>> = vec![];
+        while let Ok(Some(x)) = self.inner_assign() {
+            exprs.push(x);
+        }
```

swallows error if not Ok(Some(x))